OpenXmlPowerTools/PegBase.cs

```diff
@@ -490,373 +630,500 @@ public virtual string GetRuleNameFromId(int id)
+        }
+
+        public bool TreeCharsWithId(int nId, Matcher toMatch)
+        {
+            return TreeCharsWithId(nodeCreator_, nId, toMatch);
+        }
+
+        public bool TreeCharsWithId(Creator nodeCreator, int nId, Matcher toMatch)
+        {
+            var pos = pos_;
+            if (toMatch())
+            {
+                if (!bMute_)
+                {
+                    AddTreeNode(nId, PegTree.AddPolicy.eAddAsSibling, nodeCreator, ECreatorPhase.eCreateAndComplete);
+                    tree.cur_.match_.posBeg_ = pos;
+                    tree.cur_.match_.posEnd_ = pos_;
+                }
+                return true;
+            }
+            return false;
+        }
+
+        public bool TreeNT(int nRuleId, Matcher toMatch)
+        {
+            return TreeNT(nodeCreator_, nRuleId, toMatch);
+        }
+
+        public bool TreeNT(Creator nodeCreator, int nRuleId, Matcher toMatch)
+        {
+            if (bMute_)
+            {
+                return toMatch();
+            }
+
+            PegNode prevCur = tree.cur_, ruleNode;
+            var prevPolicy = tree.addPolicy;
+            var posBeg = pos_;
+            AddTreeNode(nRuleId, PegTree.AddPolicy.eAddAsChild, nodeCreator, ECreatorPhase.eCreate);
+            ruleNode = tree.cur_;
+            var bMatches = toMatch();
+            if (!bMatches)
+            {
+                RestoreTree(prevCur, prevPolicy);
+            }
+            else
+            {
+                ruleNode.match_.posBeg_ = posBeg;
+                ruleNode.match_.posEnd_ = pos_;
+                tree.cur_ = ruleNode;
+                tree.addPolicy = PegTree.AddPolicy.eAddAsSibling;
+                nodeCreator(ECreatorPhase.eCreationComplete, ruleNode, nRuleId);
+            }
+            return bMatches;
+        }
+
+        public bool TreeAST(int nRuleId, Matcher toMatch)
+        {
+            return TreeAST(nodeCreator_, nRuleId, toMatch);
+        }
+
+        public bool TreeAST(Creator nodeCreator, int nRuleId, Matcher toMatch)
+        {
+            if (bMute_)
+            {
+                return toMatch();
+            }
+
+            var bMatches = TreeNT(nodeCreator, nRuleId, toMatch);
+            if (bMatches)
+            {
+                if (tree.cur_.child_ != null && tree.cur_.child_.next_ == null && tree.cur_.parent_ != null)
+                {
+                    if (tree.cur_.parent_.child_ == tree.cur_)
+                    {
+                        tree.cur_.parent_.child_ = tree.cur_.child_;
+                        tree.cur_.child_.parent_ = tree.cur_.parent_;
+                        tree.cur_ = tree.cur_.child_;
+                    }
+                    else
+                    {
+                        PegNode prev;
+                        for (prev = tree.cur_.parent_.child_; prev != null && prev.next_ != tree.cur_; prev = prev.next_)
+                        {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Either remove or fill this block of code.](https://app.codacy.com/gh/Codeuctivity/Open-Xml-PowerTools/pullRequest?prid=5308324)